skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementController.java

```diff
@@ -0,0 +1,44 @@
+package com.RDS.skilltree.Endorsement;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static com.RDS.skilltree.Endorsement.JsonApiResponseConverter.convertToHashMap;
+
+@RestController
+@RequestMapping("/v1/endorsements")
+@Slf4j
+public class EndorsementController {
+    private final EndorsementService endorsementService;
+
+    public EndorsementController(EndorsementService endorsementService) {
+        this.endorsementService = endorsementService;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getEndorsementById(@PathVariable(value = "id", required = true) String id){
```

Comment:
In case of valid response the type will be `EndorsementDTO` while in case of error it will be of JSON type. Java unlike JS is not allowing to do either or in the signature hence had to use wild card `?`. Please suggest me if you know any solutions

Reply:
Currently, I have implemented the types as per your recommendations. Regarding the global exception handling, @bhtibrewal is addressing it in her pull request https://github.com/Real-Dev-Squad/skill-tree-backend/pull/41, and it is under discussion. Once her PR is merged, I plan to transition from local to global error handlers.